src/arfsdao.ts

```diff
@@ -781,4 +793,34 @@ export class ArFSDAO extends ArFSDAOAnonymous {
 		const entitiesWithPath = children.map((entity) => new ArFSPrivateFileOrFolderWithPaths(entity, hierarchy));
 		return entitiesWithPath;
 	}
+
+	async assertValidPassword(password: string): Promise<void> {
+		const wallet = this.wallet;
+		const walletAddress = await wallet.getAddress();
+		const query = buildQuery({
+			tags: [
+				{ name: 'Entity-Type', value: 'drive' },
+				{ name: 'Drive-Privacy', value: 'private' }
+			],
+			owner: walletAddress,
+			sort: ASCENDING_ORDER
+		});
+		const response = await this.arweave.api.post(graphQLURL, query);
+		const { data } = response.data;
+		const { transactions } = data;
+		const { edges } = transactions;
```

Comment:
It should early return if no drive is found. This means the user will be able to enter whichever the password it wants as it is the very first drive
```suggestion
		const { edges } = transactions;
		if (!edges.length) {
				return;
		}
```